sources/engine/Stride.Engine/Engine/Startup/GameBuilder.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Reflection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+
+namespace Stride.Engine.Startup;
+
+public interface IGameBuilder
+{
+    public ConfigurationManager Configuration { get; }
```

Comment:
ConfigurationManager standardizes application settings in one place.
They can come from a file, command line argument, environment variable, etc.

They get combined and overwrite each other. For example, ASP default is
to have appsettings.json, the values of which can be overwritten in debug
or release mode, by having appsettings.Development.json or
appsettings.Production.json respectively.

We can set up a sensible default order for these sources, but out of the box
this ConfigurationManager allows any game developer to override it as they please.

A YAML configuration provider does not exist AFAIK, but should be trivial to implement by deriving from FileConfigurationProvider
https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.fileconfigurationprovider?view=dotnet-plat-ext-7.0

Another cool thing is that one possible source is "User Secrets"
https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-7.0&tabs=windows
Which allows a game developer team member to override any configuration value
without touching anything in the project folder.

It would also be trivial to add dotnet's code for what they call "the options pattern"
https://learn.microsoft.com/en-us/dotnet/core/extensions/options
which allows an injected class to choose between getting a configuration object that
A) Just has the options from when the app started
B) Has updated values if the file was changed after app start
C) Allows to register a listener for when the file changes
Don't know if there's use for that in game dev except debugging, though.

Reply:
IIRC at least in Sdk.Web you can do it either way, and one way or the other isn't really more idiomatic than the other. So I guess it kinda comes down to the choice of also giving the users both options, or making a judgement call what to support. 

Also, just in general - as a comment that's also an answer to other things further down: In .NET 8, the generic host was made kinda... more "accessible" (?) - to the point where in my own project,  switching to the standard `IHost` allowed me to get rid of some homebrew stuff in favour of the Microsoft-provided standard.

If that's also a good fit for Stride - eh, you know it far better than me :D